### PR TITLE
Creates UserInitializer and authorizeSession functions, adds testing

### PR DESCRIPTION
**Pull request by stangah**
## Description

Refactors user initialization and authorization flow to be more readable and testable

[Pivotal Story](https://www.pivotaltracker.com/story/show/165159041)

### CODE REVIEW

**Comment by chrisgilmerproj** [comment]
fyi, if no one picks this up I may do it. I have a vested interest in the login/session flow.

**Comment by chrisgilmerproj** [comment]
I've removed Infra from this PR and added the folks from the ARBOC.

File: pkg/services/auth/user_initializer_test.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/zap"
+
+	"github.com/transcom/mymove/pkg/testingsuite"
+
+	"github.com/transcom/mymove/pkg/models"
+	"github.com/transcom/mymove/pkg/testdatagen"
+)
+
+func uuidFormatter(id uuid.UUID) *uuid.UUID {
```
Comment (lynzt) [inline pkg/services/auth/user_initializer_test.go]: rename `id` to be `uuid`? - for a quick second, I thought they were different...

**Comment by chrisgilmerproj** [comment]
Closing since this hasn't been picked up for work.

**Pull request closed without merging**